x-pack/plugins/osquery/public/scheduled_query_groups/queries/platform_checkbox_group_field.tsx

```diff
@@ -112,6 +112,15 @@ export const PlatformCheckBoxGroupField = ({
 
   const describedByIds = useMemo(() => (idAria ? [idAria] : []), [idAria]);
 
+  useEffect(() => {
+    setCheckboxIdToSelectedMap(() =>
+      (options as EuiCheckboxGroupOption[]).reduce((acc, option) => {
+        acc[option.id] = isEmpty(field.value) ? true : field.value?.includes(option.id) ?? false;
```

Comment:
a `Set` seems like it would be a good choice for this, something like:
```
setCheckboxIdToSelectedSet(
  new Set(
    isEmpty(field.value) 
    ? options.map(op => op.id) 
    : field.value
  )
)
```

this will create an empty set in the case where `field.value` is `null`

i'm not sure what the performance implications are here, so i'll mark it as a nit